modules/renter/contractor/editor.go

```diff
@@ -145,6 +151,21 @@ func (c *Contractor) Editor(contract modules.RenterContract) (Editor, error) {
 		}
 	}
 
+	// acquire revising lock
+	c.mu.Lock()
+	alreadyRevising := c.revising[contract.ID]
+	if alreadyRevising {
+		c.mu.Unlock()
+		return nil, errors.New("already revising that contract")
```

Comment:
a `sync.Cond` might be appropriate here. That would allow us to efficiently wait for the revision to become available. You get the nice blocking properties without having to spawn background loops that send `Editor`s down channels and whatnot.